### PR TITLE
Added model and function to generate unique human readable identifica…

### PR DESCRIPTION
**Pull request by stevenbal**
…tion for resources

### CODE REVIEW

**Comment by stevenbal** [comment]
needed for https://github.com/VNG-Realisatie/gemma-zaakregistratiecomponent/pull/86

File: vng_api_common/utils.py [hunk 1]
```diff
@@ -110,3 +110,19 @@ def request_object_attribute(url: str, attribute: str, resource: Union[str, None
         logger.warning("%s was retrieved from %s with the %s: %s", attribute, url, exc.__class__.__name__, exc)
         result = ''
     return result
+
+
+def generate_unique_identification(instance, date_field_name):
+    from .models import IssuedIdentifications
+    model_name = instance._meta.model_name.upper()
+    year = instance.registratiedatum.year
```
Comment (sergei-maertens) [inline vng_api_common/utils.py]: you're not using the `date_field_name` parameter, but hardcoding here

File: vng_api_common/utils.py [hunk 2]
```diff
@@ -110,3 +110,19 @@ def request_object_attribute(url: str, attribute: str, resource: Union[str, None
         logger.warning("%s was retrieved from %s with the %s: %s", attribute, url, exc.__class__.__name__, exc)
         result = ''
     return result
+
+
+def generate_unique_identification(instance, date_field_name):
```
Comment (sergei-maertens) [inline vng_api_common/utils.py]: Please add typehints to make it easier to reason about the code :smiley:

File: vng_api_common/utils.py [hunk 3]
```diff
@@ -110,3 +110,19 @@ def request_object_attribute(url: str, attribute: str, resource: Union[str, None
...
+    from .models import IssuedIdentifications
```
Comment (sergei-maertens) [inline vng_api_common/utils.py]: why is this a local import?

File: vng_api_common/utils.py (hunk 1 above)
Comment (stevenbal) [reply vng_api_common/utils.py]: whoops

File: vng_api_common/utils.py [hunk 4]
```diff
@@ -110,3 +110,22 @@ def request_object_attribute(url: str, attribute: str, resource: Union[str, None
...
+def generate_unique_identification(instance: models.Model, date_field_name: str):
+    model_name = instance._meta.model_name.upper()
+    year = getattr(instance, date_field_name).year
+    prefix = f'{model_name}-{year}'
+
+    pattern = prefix + r'-\d{10}'
+
+    issued_ids_for_year = instance.__class__.objects.filter(identificatie__regex=pattern)
+
+    if issued_ids_for_year:
```
Comment (sergei-maertens) [inline vng_api_common/utils.py]: Avoid loading the entire dataset into memory:

```suggestion
    if issued_ids_for_year.exists():
```

File: vng_api_common/utils.py [hunk 5]
```diff
@@ -110,3 +110,22 @@ def request_object_attribute(url: str, attribute: str, resource: Union[str, None
...
+
+
+def generate_unique_identification(instance: models.Model, date_field_name: str):
+    model_name = instance._meta.model_name.upper()
+    year = getattr(instance, date_field_name).year
+    prefix = f'{model_name}-{year}'
+
+    pattern = prefix + r'-\d{10}'
+
+    issued_ids_for_year = instance.__class__.objects.filter(identificatie__regex=pattern)
```
Comment (sergei-maertens) [inline vng_api_common/utils.py]: Django marks the first defined manager as 'default' so you don't have to rely on the manager being named `objects`

```suggestion
    issued_ids_for_year = instance.__class__._default_manager.filter(identificatie__regex=pattern)
```

**Comment by joeribekker** [comment]
Had van mij niet op model naam gehoeven, en had pluggable (zoals bij Haarlem) kunnen zijn maar prima voor nu.

**Pull request merged**